src/ameba/rule/lint/percent_array.cr

```diff
@@ -42,10 +42,12 @@ module Ameba::Rule::Lint
           start_token = token.dup
         when .string?
           if start_token && issue.nil?
+            # ameba:disable Lint/NotNil
             issue = array_entry_invalid?(token.value, start_token.not_nil!.raw)
           end
```

Comment:
can this case be handled by adding an extra variable? i.e.

_sorry, hadn't have a chance to give it a try_

```suggestion
          if (start = start_token) && issue.nil?
            issue = array_entry_invalid?(token.value, start.raw)
          end
```

Reply:
It could be done, but it changes semantics and makes the code more convoluted. In this case I feel that using `not_nil!` is more apt, kinda exception that proves the rule ;)

Reply:
Well, maybe not... I've pushed the changes with local variables.